sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/fcm/RegistrationKeyStore.java

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.android.communication.chat.implementation.notifications.fcm;
+
+import android.util.Log;
+
+import com.azure.android.core.logging.ClientLogger;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class represents a storage facility for secrete key as string and its creation time. The
+ * data is persisted in a file as json format. The key-store itself could load data from file for
+ * faster operation.
+ *
+ * The file persists all the data in Json format. For example:
+ * {
+ *     "alias_1": {
+ *         "credential": "abcdefgh",
+ *         "creationTime": "1234";
+ *     },
+ *     "alias_2": {
+ *         "credential": "abcdefgh",
+ *         "creationTime": "1234";
+ *     }
+ * }
+ */
+public class RegistrationKeyStore {
+    private Map<String, RegistrationKeyEntry> map;
```

Comment:
Not sure why an alias is needed. Can this be just a list of keys?

Reply:
The major reason is that we need to distinguish cryptoKeys and authKeys. Having a key with different prefix is an easy way to achieve this. Using list structure probably requires managing separate files for each of them.

Reply:
aren't the crypto and auth keys always go in pairs and share the same expiry time? if so, why don't we have a structure like this:
```json
[
  { "crypt": "...", "auth": "...", expiryTime: ... },
  { "crypt": "...", "auth": "...", expiryTime: ... },
  { "crypt": "...", "auth": "...", expiryTime: ... }
]
```

Reply:
Let me trying implementing this

Reply:
Changed to list